### DISCUSSION TITLE
Feature Request Traefik Proxy Support for IngressRouteTCP

### DISCUSSION THREAD

**Discussion by kulaseba** [argoproj/argo-rollouts#4024] [Q&A]
Hello [Argo Rollouts Support/Community],

I would like to propose a feature enhancement regarding the support for `TraefikService` in conjunction with `IngressRouteTCP`.

As of now, Argo Rollouts seamlessly integrates with `TraefikService`, allowing users to manage traffic weights and load balancing strategies effectively when deploying applications through `IngressRoute`. However, it is noted that `IngressRouteTCP` does not currently support a similar functionality with `TraefikService`.

https://argo-rollouts.readthedocs.io/en/stable/features/traffic-management/traefik/

### Proposed Enhancement

I would like to request that support for `TraefikService` be extended to `IngressRouteTCP`. Here are a few reasons why this enhancement would be beneficial:

1. **Consistent Configuration Options**:
   - Enabling `TraefikService` to work with `IngressRouteTCP` would provide a consistent configuration experience across different types of ingress resources, allowing users to manage TCP services in a similar manner to HTTP services.

2. **Enhanced Load Balancing**:
   - Adding `TraefikService` support for `IngressRouteTCP` would empower users to define custom load balancing strategies, health checks, and service configurations specifically for TCP traffic, thereby improving resilience and performance.

3. **Simplified Management**:
   - Consolidating traffic management settings for both HTTP and TCP services allows for easier maintenance and understanding of configuration, reducing operational complexity.

4. **Improved Traffic Control**:
   - The ability to implement advanced traffic control mechanisms through `TraefikService` on TCP routes would enhance deployment strategies, especially for canary deployments and blue/green strategies.

### Conclusion

Providing support for `TraefikService` in `IngressRouteTCP` would significantly enhance the flexibility and functionality of Argo Rollouts when managing TCP-based applications. I believe this feature request aligns well with the ongoing commitment to improving the user experience of Argo Rollouts.

Thank you for considering this request. I look forward to hearing your thoughts on the feasibility of this enhancement.

Best regards,
Balakrishnan Kulasekaran

**Comment by kostis-codefresh** [comment]
It would be much easier to try using the TCP support of the Gateway API plugin https://rollouts-plugin-trafficrouter-gatewayapi.readthedocs.io/en/latest/features/tcp/

Traefik implements the Gateway API plugin just fine.

**Reply by kulaseba** [reply]
@kostis-codefresh : I got your thinking. In my usecase we are in a situation to handle Http services as well which IngressRouteTCP does alongside.TCPRoute is more focused and flexible for defining TCP routing rules alone
